api/CIXApi.ts

```diff
@@ -0,0 +1,281 @@
+    constructor(baseUrl: string, handlers: VsoBaseInterfaces.IRequestHandler[], options?: VsoBaseInterfaces.IRequestOptions) {
+        super(baseUrl, handlers, 'node-Pipelines-api', options);
+    }
+
+    /**
+     * Gets a list of existing configuration files for the given repository.
+     * 
+     * @param {string} project - Project ID or project name
+     * @param {string} repositoryType - The type of the repository such as GitHub, TfsGit (i.e. Azure Repos), Bitbucket, etc.
+     * @param {string} repositoryId - The vendor-specific identifier or the name of the repository, e.g. Microsoft/vscode (GitHub) or e9d82045-ddba-4e01-a63d-2ab9f040af62 (Azure Repos)
+     * @param {string} branch - The repository branch where to look for the configuration file.
+     * @param {string} serviceConnectionId - If specified, the ID of the service endpoint to query. Can only be omitted for providers that do not use service endpoints, e.g. TfsGit (i.e. Azure Repos).
+     */
+    public async getConfigurations(
+        project: string,
+        repositoryType?: string,
+        repositoryId?: string,
+        branch?: string,
+        serviceConnectionId?: string
+        ): Promise<CIXInterfaces.ConfigurationFile[]> {
+
+        return new Promise<CIXInterfaces.ConfigurationFile[]>(async (resolve, reject) => {
+            let routeValues: any = {
+                project: project
+            };
+
+            let queryValues: any = {
+                repositoryType: repositoryType,
+                repositoryId: repositoryId,
+                branch: branch,
+                serviceConnectionId: serviceConnectionId,
+            };
+            
+            try {
+                let verData: vsom.ClientVersioningData = await this.vsoClient.getVersioningData(
+                    "7.2-preview.1",
+                    "pipelines",
+                    "8fc87684-9ebc-4c37-ab92-f4ac4a58cb3a",
+                    routeValues,
+                    queryValues);
+
+                let url: string = verData.requestUrl!;
+                let options: restm.IRequestOptions = this.createRequestOptions('application/json', 
+                                                                                verData.apiVersion);
+
+                let res: restm.IRestResponse<CIXInterfaces.ConfigurationFile[]>;
+                res = await this.rest.get<CIXInterfaces.ConfigurationFile[]>(url, options);
+
+                let ret = this.formatResponse(res.result,
+                                              null,
+                                              true);
+
+                resolve(ret);
+                
+            }
+            catch (err) {
+                reject(err);
+            }
+        });
+    }
+
+    /**
+     * LEGACY METHOD - Obsolete Creates a new Pipeline connection between the provider installation and the specified project. Returns an Operation object that holds the redirect Url
```

Comment:
Do we need to add and support legacy/obsolete methods?

Reply:
Removed legacy method, updated doc, thanks!